adapters/nsw.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+var request = require('request');
+var _ = require('lodash');
+var moment = require('moment-timezone');
+var cheerio = require('cheerio');
+
+exports.name = 'nsw';
+
+exports.fetchData = function (source, cb) {
+  request(source.url, function (err, res, body) {
+    if (err || res.statusCode !== 200) {
+      console.error(err || res);
+      return cb({message: 'Failure to load data url.'});
+    }
+
+    // Wrap everything in a try/catch in case something goes wrong
+    try {
+      // Format the data
+      var data = formatData(body);
+      // Make sure the data is valid
+      if (data === undefined) {
+        return cb({message: 'Failure to parse data.'});
+      }
+      cb(null, data);
+    } catch (e) {
+      return cb({message: 'Unknown adapter error.'});
+    }
+  });
+};
+
+// This is pretty nasty, but the rest of the code is as well.
+var parseDate = function (dateString) {
+  var d = dateString.split('<br>');
+  // Split the hour string to get starting hour + am/pm
+  var h = d[2].split(' ');
+  var date = moment.tz(d[1] + h[0] + h[3], 'D MMMM YYYYha', 'Australia/Melbourne');
```

Comment:
I would argue 4pm because that's the hour it completed and when, I would
assume, the data is released publicly (or, you know, 4:02pm). I think as
long as it is clearly defined it doesn't matter though.

On Wed, Oct 7, 2015 at 5:29 PM, Joe Flasher <notifications@github.com>
wrote:

> In adapters/nsw.js
> <https://github.com/openaq/openaq-api/pull/80#discussion_r41450174>:
>
> > +      if (data === undefined) {
> > +        return cb({message: 'Failure to parse data.'});
> > +      }
> > +      cb(null, data);
> > +    } catch (e) {
> > +      return cb({message: 'Unknown adapter error.'});
> > +    }
> > +  });
> > +};
> > +
> > +// This is pretty nasty, but the rest of the code is as well.
> > +var parseDate = function (dateString) {
> > +  var d = dateString.split('<br>');
> > +  // Split the hour string to get starting hour + am/pm
> > +  var h = d[2].split(' ');
> > +  var date = moment.tz(d[1] + h[0] + h[3], 'D MMMM YYYYha', 'Australia/Melbourne');
>
> @olafveerman <https://github.com/olafveerman> good point, @RocketD0g
> <https://github.com/RocketD0g> any input here? If a time says 3-4 pm,
> which value should we be using for the time that's stored with the
> measurement?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openaq/openaq-api/pull/80/files#r41450174>.
>



-- 

507.246.2097